### DISCUSSION TITLE
Converting from UK Grid Ref to WGS84 (for Google maps API)

### DISCUSSION THREAD

**Discussion by stevegoddard** [dvdoug/PHPCoord#35] [Q&A] [answered]
Hi,
I'm updating some code to version 4 of PHPCoord.
Most of what I do revolves around UK grid references being converted to lat / lng (WGS84) for display on Bing and Google maps APIs.
In the past (version 3)  I would do something like this:

    $osref = OSRef::fromGridReference($gridref);

    $easting = $osref->getX();
    $northing = $osref->getY();

    $lat_lng = $osref->toLatLng();
    $lat_lng = $lat_lng->toWGS84();
    $lat = $lat_lng->getLat();
    $lng = $lat_lng->getLng();

It was the lat and lng I was most interested in. The accuracy seemed to be very good - within a few meters.

With version 4, i'm doing something like this:

    $point = BritishNationalGridPoint::fromGridReference($gridref);
    $to_crs = Geographic2D::fromSRID(Geographic2D::EPSG_OSGB_1936);
    $to = $point->convert($to_crs);
    $lat = floatval((string) $to->getLatitude());
    $lng = floatval((string) $to->getLongitude());

This works, but i'd say the result (for mid-Wales, UK) is about 1-200meters out.

Any suggestions what i'm doing wrong? Am I using the wrong CRS or am I missing a conversion step?

Thanks
Steve

**Comment by dvdoug** [comment, marked as answer]
Hello

You're going from OSGB36 (National Grid)...to OSGB36 with that code. I think you probably want `$to_crs = Geographic2D::fromSRID(Geographic2D::EPSG_WGS84);`

Side note: Out of the box v4 will use a Helmert transform for this (like v3 did). In the next few days (next release), it will be able to use the definitive OSTN15 transform instead if `php-coord/datapack-europe` is also installed.

**Reply by stevegoddard** [reply]
Argh... that was as silly mistake!
Many thanks - working fine now!
